app/controllers/facts_controller.rb

```diff
@@ -2,18 +2,42 @@ class FactsController < ApplicationController
   before_action :authenticate_user!, except: [:index, :show]
 
   def index
-    @tables = []
     titles = Title.all
     years = Year.all
-    titles.each do |title|
-      @tables << title.facts
-    end
     @t = []
     titles.each do |title|
       years.each do |year|
         @t << Fact.where(title: title, year: year)
       end
     end
+    @tables = {}
+    titles.each do |title|
+      uniq_years = title.years.map{|x| x.year}.uniq.sort
+      uniq_locations = title.locations.map{|x| x.country}.uniq
```

Comment:
Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.